src/policy/fees.cpp

```diff
@@ -690,6 +690,20 @@ void CBlockPolicyEstimator::processBlock(const std::vector<RemovedMempoolTransac
     shortStats->UpdateMovingAverages();
     longStats->UpdateMovingAverages();
 
+    std::set<Txid> seen_transactions;
+    for (const auto& tx : txs_removed_for_block) {
+        seen_transactions.insert(tx.info.m_tx->GetHash());
+        for (const auto& input : tx.info.m_tx->vin) {
+            const Txid& parentId = input.prevout.hash;
+            if (seen_transactions.count(parentId)) {
```

Comment:
I'm assuming the transactions are sorted from parents to children. So in the case A -> B -> C -> D, all except D would be removed, right?

Reply:
>  I'm assuming the transactions are sorted from parents to children.

Yes, they are.

The fee estimator does not take all transaction with parents into account already see
https://github.com/bitcoin/bitcoin/blob/7fcf4e99791ca5be0b068ac03a81a50ece11dba3/src/policy/fees.cpp#L615

Assuming A->B->C->D are topologically sorted from parent to descendant.

In this case B, C, D are already not considered for fee estimation, After this PR A will also be ignored.

Reply:
Hmm I see, as per this comment it says "not dependent on any other transactions in the **mempool**" - does this also apply to other transactions in the **block**?

https://github.com/bitcoin/bitcoin/blob/7fcf4e99791ca5be0b068ac03a81a50ece11dba3/src/policy/fees.cpp#L613

Reply:
> does this also apply to other transactions in the block?

I don't fully understand your question. 
The comment you highlighted refers to all incoming transactions into the mempool.

After a new block arrives, the fee estimator listens for notifications of all transactions removed from mempool because of the new block, and updates their tracking starts. All transactions with mempool dependencies that are mined are not tracked initially, they are just going to be skipped.